sim/pkg/simulator/array_stock.go

```diff
@@ -0,0 +1,94 @@
+package simulator
+
+import "fmt"
+
+type arrayStock struct {
+	name       StockName
+	stocksKind EntityKind
+	stock      []*Entity
+}
+
+func (as *arrayStock) Name() StockName {
+	return as.name
+}
+
+func (as *arrayStock) KindStocked() EntityKind {
+	return as.stocksKind
+}
+
+func (as *arrayStock) Count() uint64 {
+	return uint64(len(as.stock))
+}
+
+//time complexity O(len(as.stock))
```

Comment:
Isn't this O(1)?  Because you're just returning the stock slice as-is.

Reply:
Sure! It's O(1). copy-paste error